packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -709,3 +717,99 @@ class _CupertinoEdgeShadowPainter extends BoxPainter {
     canvas.drawRect(rect, paint);
   }
 }
+
+class _CupertinoModalPopupRoute<T> extends PopupRoute<T> {
+  _CupertinoModalPopupRoute({
+    this.builder,
+    this.barrierLabel,
+    RouteSettings settings,
+  }) : super(settings: settings);
+
+  final WidgetBuilder builder;
+
+  @override
+  final String barrierLabel;
+
+  @override
+  Color get barrierColor => _kModalBarrierColor;
+
+  @override
+  bool get barrierDismissible => true;
+
+  @override
+  bool get semanticsDismissible => false;
+
+  @override
+  Duration get transitionDuration => _kModalPopupTransitionDuration;
+
+  Animation<double> _animation;
+
+  @override
+  Animation<double> createAnimation() {
+    assert(_animation == null);
+    _animation = new CurvedAnimation(
+      parent: super.createAnimation(),
+      curve: Curves.ease,
+      reverseCurve: Curves.ease.flipped,
+    );
+    return _animation;
+  }
+
+  @override
+  Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+    return builder(context);
+  }
+
+  @override
+  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
+    if (_animation.value == 1.0) {
+      context.findRenderObject().markNeedsSemanticsUpdate();
+    }
+    return new Align(
+      alignment: Alignment.bottomCenter,
+      child: new FractionalTranslation(
+        translation: new Tween<Offset>(
+          begin: const Offset(0.0, 1.0),
+          end: const Offset(0.0, 0.0),
+        ).evaluate(_animation),
+        child: child,
+      ),
+    );
+  }
+}
+
+/// Shows a modal iOS-style popup that slides up from the bottom of the screen.
+///
+/// Such a popup is an alternative to a menu or a dialog and prevents the user
+/// from interacting with the rest of the app.
+///
+/// The `context` argument is used to look up the [Navigator] for the popup.
+/// It is only used when the method is called. Its corresponding widget can be
+/// safely removed from the tree before the popup is closed.
+///
+/// The `builder` argument typically builds a [ActionSheet] widget. Content
+/// below the widget is dimmed with a [ModalBarrier]. The widget built by the
+/// `builder` does not share a context with the location that
+/// `showCupertinoModalPopup` is originally called from. Use a
+/// [StatefulBuilder] or a custom [StatefulWidget] if the widget needs to
+/// update dynamically.
+///
+/// Returns a `Future` that resolves to the value that was passed to
+/// [Navigator.pop] when the popup was closed.
+///
+/// See also:
+///
+///  * [ActionSheet], which is the widget usually returned by the `builder`
```

Comment:
Should ActionSheet be CupertinoActionSheet? Or is that all happening in the other rename PR?